src/main/java/ch/njol/skript/events/EvtClick.java

```diff
@@ -142,82 +124,86 @@ public boolean check(Event event) {
-				if (!interactTracker.checkEvent(clickEvent.getPlayer(), clickEvent, clickEvent.getHand())) {
+				if (!interactTracker.checkEvent(interactEntityEvent.getPlayer(), interactEntityEvent, interactEntityEvent.getHand())) {
 					return false; // Not first event this tick
 				}
 			}
 			
 			entity = clicked;
 			block = null;
-		} else if (event instanceof PlayerInteractEvent) {
-			PlayerInteractEvent clickEvent = ((PlayerInteractEvent) event);
-			
+		} else if (event instanceof PlayerInteractEvent interactEvent) {
 			// Figure out click type, filter non-click events
-			Action a = clickEvent.getAction();
+			Action action = interactEvent.getAction();
 			int click;
-			switch (a) {
-				case LEFT_CLICK_AIR:
-				case LEFT_CLICK_BLOCK:
-					click = LEFT;
-					break;
-				case RIGHT_CLICK_AIR:
-				case RIGHT_CLICK_BLOCK:
-					click = RIGHT;
-					break;
-				case PHYSICAL: // Not a click event
-				default:
+			switch (action)  {
+				case LEFT_CLICK_AIR, LEFT_CLICK_BLOCK -> click = LEFT;
+				case RIGHT_CLICK_AIR, RIGHT_CLICK_BLOCK -> click = RIGHT;
+				case PHYSICAL -> {
+					return false;
+				}
+				default -> {
 					return false;
+				}
 			}
 			if ((this.click & click) == 0)
 				return false; // We don't want to handle this kind of events
 			
-			EquipmentSlot hand = clickEvent.getHand();
+			EquipmentSlot hand = interactEvent.getHand();
 			assert hand != null; // Not PHYSICAL interaction
-			if (!interactTracker.checkEvent(clickEvent.getPlayer(), clickEvent, hand)) {
+			if (!interactTracker.checkEvent(interactEvent.getPlayer(), interactEvent, hand)) {
 				return false; // Not first event this tick
 			}
 			
-			block = clickEvent.getClickedBlock();
+			block = interactEvent.getClickedBlock();
 			entity = null;
 		} else {
 			assert false;
 			return false;
 		}
-		
-		if (tools != null && !tools.check(event, new Checker<ItemType>() {
-			@Override
-			public boolean check(final ItemType t) {
-				if (event instanceof PlayerInteractEvent) {
-					return t.isOfType(((PlayerInteractEvent) event).getItem());
-				} else { // PlayerInteractEntityEvent doesn't have item associated with it
-					PlayerInventory invi = ((PlayerInteractEntityEvent) event).getPlayer().getInventory();
-					ItemStack item = ((PlayerInteractEntityEvent) event).getHand() == EquipmentSlot.HAND
-							? invi.getItemInMainHand() : invi.getItemInOffHand();
-					return t.isOfType(item);
-				}
+
+		if (tools != null && !tools.check(event, itemType -> {
+			if (event instanceof PlayerInteractEvent interactEvent) {
+				return itemType.isOfType(interactEvent.getItem());
+			} else {
+				PlayerInventory invi = ((PlayerInteractEntityEvent) event).getPlayer().getInventory();
+				ItemStack item = ((PlayerInteractEntityEvent) event).getHand() == EquipmentSlot.HAND
+						? invi.getItemInMainHand() : invi.getItemInOffHand();
+				return itemType.isOfType(item);
 			}
-		})) {
-			return false;
-		}
-		
+		})) { return false; }
```

Comment:
```suggestion
		})) {
			return false;
		}
```

Reply:
actually you might want to declare the lambda as a predicate prior to the if statement, so the if statement doesn't get lost in the sauce.